addon/macros/intl.js

```diff
@@ -0,0 +1,24 @@
+import { computed, get, defineProperty } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export const __intlInjectionName = `intl-${Date.now().toString(36)}`;
+
+export default function intl(...dependentKeysAndGetterFn) {
+  const getterFn = dependentKeysAndGetterFn.pop();
+  const dependentKeys = dependentKeysAndGetterFn;
+
+  return computed(function(propertyKey) {
+    if (!get(this, __intlInjectionName)) {
+      defineProperty(this, __intlInjectionName, service('intl'));
```

Comment:
It might be cleaner to do `this.set(__intlInjectionName, getOwner(this).lookup('service:intl'));`, this is a case where using the owner directly would actually probably be cheaper and a bit cleaner.

Reply:
Oh yes, definitely!